File: http-streams/src/main/scala/http/Routing.scala
package sc.http

import akka.NotUsed
import akka.actor.ActorSystem
import akka.http.scaladsl.Http
import akka.http.scaladsl.model.HttpRequest
import akka.http.scaladsl.model.HttpResponse
import akka.http.scaladsl.server.{Directives, Route, RouteResult}
import akka.stream.ActorMaterializer
import akka.stream.scaladsl.Flow

object Routing extends Directives {
  implicit val system = ActorSystem()
  val http = Http()
  implicit val materializer = ActorMaterializer()
  //#simple
  val route: Route = path("hello") {
    get {
      complete("Hello, world!")
    }
  }
  //#simple
  //#run
  // RouteResult.route2HandlerFlow(route)
  val flow2: Flow[HttpRequest, HttpResponse, NotUsed] = route
  //#run
}

File: http-streams/src/main/java/http/Routing.java
package http;

import akka.NotUsed;
import akka.actor.ActorSystem;
import akka.http.javadsl.Http;
import akka.http.javadsl.model.HttpRequest;
import akka.http.javadsl.model.HttpResponse;
import akka.http.javadsl.server.*;
import akka.stream.ActorMaterializer;
import akka.stream.Materializer;
import akka.stream.javadsl.Flow;

import java.util.function.BiFunction;
import java.util.function.Supplier;

public class Routing extends AllDirectives {

  void run() {
    ActorSystem system = ActorSystem.create("routes");

    Http http = Http.get(system);
    Materializer materializer = ActorMaterializer.create(system);

    //#simple
    Route route = path("hello", () ->
      get(() ->
        complete("Hello, world!")
      )
    );
    //#simple
    //#run
    Flow<HttpRequest, HttpResponse, NotUsed> flow =
      route.flow(system, materializer);
    //#run
  }
}
